grippers/rr_suction_pump/include/rr_suction_pump/suction_pump.hpp

```diff
@@ -69,6 +59,37 @@ class Pump
         }
     };
 
+protected:
+    std::vector<HI_ptr> _trigger;
+    bool _outputNormallyOn;
+    bool _inputNormallyOn;
+};
+
+template <class HI, class Config>
+class PumpWithFeedback : public Pump<HI, Config>
+{
+public:
+    typedef std::unique_ptr<HI> HI_ptr;
+    PumpWithFeedback(const std::vector<std::string> triggerPins, const int numOfTrigger, const std::vector<std::string> statusPins, const int numOfStatus,
+            const bool outputNormallyOn, const bool inputNormallyOn)
+            : Pump<HI, Config>(triggerPins, numOfTrigger, outputNormallyOn), _inputNormallyOn(inputNormallyOn)
+    {
+        for (int i = 0; i < numOfStatus; i++) {
+            _status.push_back(HI_ptr(new HI(statusPins[i], HI::Type::RR_HW_INTERFACE_INPUT)));
```

Comment:
```suggestion
            _status.emplace_back(new HI(statusPins[i], HI::Type::RR_HW_INTERFACE_INPUT));
```
saves a copy and text if you care (similar with other places)